pydantic/main.py

```diff
@@ -366,6 +366,16 @@ def __setattr__(self, name, value):
         elif self.__config__.validate_assignment:
             known_field = self.__fields__.get(name, None)
             if known_field:
+                new_values = {**self.__dict__, name: value}
+
+                for skip_on_failure, validator in self.__post_root_validators__:
```

Comment:
this should be called after validating the field and not all `skip_on_failure` validators should be skipped.

Reply:
so it's more like standard validation.